src/bin/ion/transcribe.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::{bail, Result};
+use ion_rs::*;
+use std::io::Write;
+
+/// Constructs the appropriate writer for the given format, then writes all values from the
+/// `Reader` to the new `Writer`.
+pub(crate) fn write_all_as<I: IonInput>(
+    reader: &mut Reader<AnyEncoding, I>,
+    output: &mut impl Write,
+    format: &str,
+) -> Result<usize> {
+    write_n_as(reader, output, format, usize::MAX)
+}
```

Comment:
🗺️ This module houses logic that was common to both `head` and `cat`, addressing issue #49.